hbase-client/src/main/java/org/apache/hadoop/hbase/client/HBaseAdmin.java

```diff
@@ -1717,7 +1716,13 @@ public void mergeRegionsSync(final byte[] nameOfRegionA, final byte[] nameOfRegi
   @Override
   public void mergeRegions(final byte[] nameOfRegionA, final byte[] nameOfRegionB,
     final boolean forcible) throws IOException {
-    mergeRegionsAsync(nameOfRegionA, nameOfRegionB, forcible);
+    try {
```

Comment:
`mergeRegions` will return immediately without waiting for or surfacing any exception from the operation. This seems wrong. Yet, it is a semantic change over previously released versions. For discussion.

Reply:
IIRC this is a known issue. You can see the comments in the Admin interface.

```
  /**
   * Merge two regions. Asynchronous operation.
   * @param nameOfRegionA encoded or full name of region a
   * @param nameOfRegionB encoded or full name of region b
   * @param forcible      <code>true</code> if do a compulsory merge, otherwise we will only merge
   *                      two adjacent regions
   * @throws IOException if a remote or network exception occurs
   * @deprecated Since 2.0. Will be removed in 3.0. Use
   *             {@link #mergeRegionsAsync(byte[], byte[], boolean)} instead.
   */
  @Deprecated
  void mergeRegions(byte[] nameOfRegionA, byte[] nameOfRegionB, boolean forcible)
    throws IOException;
```

Reply:
I see, I will ignore these cases then.